src/ConstructiveSolidGeometry/Intervals.jl

```diff
@@ -15,4 +15,80 @@
 @inline _right_radial_interval(r::AbstractInterval) = r.right
 
 @inline _extend_number_to_zero_interval(r::T) where {T <: Real} = T(0)..r
-@inline _extend_number_to_zero_interval(r::AbstractInterval) = r
\ No newline at end of file
+@inline _extend_number_to_zero_interval(r::AbstractInterval) = r
+
+@inline get_angular_interval(::Type{T}, α::AbstractInterval{T}) where {T} = α
+@inline get_angular_interval(::Type{T}, α::Nothing) where {T} = T(0)..T(2π)
+
+@inline _in_angular_interval_closed(α::Real, α_int::Nothing) = true
+@inline _in_angular_interval_closed(α::Real, α_int::Nothing, tol::Real) = true
+@inline _in_angular_interval_closed(α::Real, α_int::AbstractInterval{T}) where {T} = mod(α - α_int.left, T(2π)) ≤ α_int.right - α_int.left
+@inline _in_angular_interval_closed(α::Real, α_int::AbstractInterval{T}, tol::Real) where {T} = mod(α - (α_int.left-tol), T(2π)) ≤ (α_int.right+tol) - (α_int.left-tol)
+
+@inline _in_angular_interval_open(α::T, α_int::Nothing) where {T<:Real} = 0 < mod(α, T(2π)) < 2π
```

Comment:
`_in_angular_interval_open` is not used for now. But it will be needed for surface interactions. Cutting angular intervals is tricky because with just one cut you cannot guarantee you will end up with 2 intervals. With full `2π` intervals you need two cuts which are neither at 0 or `2π` to get something that is actually cut.